openc3/lib/openc3/models/target_model.rb

```diff
@@ -257,6 +257,20 @@ def self.packet_items(target_name, packet_name, items, type: :TLM, scope:)
       found
     end
 
+    # @return [Array<Hash>] All the item hashes for every packet in a target
+    def self.all_items_metadata(target_name, type: :TLM, scope:)
+      items = []
+      packets = packets(target_name, type: type, scope: scope)
+      items += packets.flat_map do |packet|
+        packet['items'].map do |item|
+          item.slice('name', 'description', 'data_type', 'array_size', 'bit_size')
+        end
+      end
+      items.uniq! { |i| i['name'] } # since this would be used by `<target_name> LATEST <item_name>`
+      items.sort_by! { |i| i['name'] }
+      items
+    end
```

Comment:
Not sure how this scales with packets with thousands of items?
@ryanmelt any concerns about the performance of this?

Reply:
We should cache this list in Redis.  Build on plugin install, and here if not already defined.